components/validator/challenge/fetcher.go

```diff
@@ -82,3 +81,51 @@ func Decode(data []byte) []*big.Int {
 
 	return result
 }
+
+var _ ProverClient = (*JsonRPCProverClient)(nil)
+
+type JsonRPCProverClient struct {
+	address string
+}
+
+func (c JsonRPCProverClient) Prove(ctx context.Context, traceString string, proofType Type) (*ProveResponse, error) {
+	reqBody := struct {
+		Jsonrpc string `json:"jsonrpc"`
+		Method  string `json:"method"`
+		Params  any    `json:"params"`
+		Id      string `json:"id"`
+	}{
+		Jsonrpc: "2.0",
+		Method:  "prove",
+		Params:  []any{traceString, proofType},
+		Id:      "0",
+	}
+
+	reqBytes, err := json.Marshal(reqBody)
+	if err != nil {
+		return nil, fmt.Errorf("failed to json.Marshal %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, c.address, bytes.NewReader(reqBytes))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new reqBody for proof: %w", err)
+	}
+
+	cli := http.Client{}
```

Comment:
How about handle timeout handling for HTTP requests at the HTTP protocol level?
```suggestion
	cli := http.Client{Timeout: f.timeout}
```

Reply:
I wanted to cancel the request when parent context is done.
So I used child context with timeout.